src/main/java/subway/line/domain/entity/LineSections.java

```diff
@@ -0,0 +1,78 @@
+package subway.line.domain.entity;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import subway.common.exception.SubwayException;
+import subway.common.exception.SubwayExceptionType;
+import subway.line.exception.InvalidDownStationException;
+import subway.line.exception.InvalidUpStationException;
+import subway.station.domain.Station;
+
+@Embeddable
+@Getter
+@NoArgsConstructor
+public class LineSections {
+
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, fetch = FetchType.LAZY, orphanRemoval = true)
```

Comment:
<img width="1263" alt="스크린샷 2024-07-20 오후 1 50 31" src="https://github.com/user-attachments/assets/f0f0db21-0169-4f56-bba2-335c047bb81a">

현재 테스트를 수행해보면 이런 에러가 발생합니다 😢 

```suggestion
    @OneToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE}, fetch = FetchType.LAZY, orphanRemoval = true)
    @JoinColumn(name = "line_id")
```
와 같이 수정하거나 

```java
@OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, fetch = FetchType.LAZY, orphanRemoval = true)
```
으로 만들고 LineSection 에서 Line을 가지도록
```java
@ManyToOne
@JoinColumn(name = "line_id")
private Line line;
```

같은 형태로 고쳐 주시는 것이 좋습니다 

1번과 같이 mappedBy 삭제 후 joincolumn 을 추가하면, 1:N 단방향 매핑이 되는 데 이것은 불필요한 update 를 발생 시킬 수 있습니다.
2번과 같이 설정하는 것은 1:N / N:1 양방향 매핑입니다.
각각에 대한 차이점은 따로 학습해보시면 좋을 것 같습니다 😄

Reply:
양방향 매핑으로 수정하면서 @ManyToOne 어노테이션을 누락한 것 같습니다 😭
개선하도록 하겠습니다!